std/traits.d

```diff
@@ -3554,6 +3555,176 @@ enum hasMember(T, string name) = __traits(hasMember, T, name);
 }
 
 /**
+ * Whether the symbol represented by the string, member, is a static member of T.
+ *
+ * Params:
+ *     T = Type containing symbol $(D member).
+ *     member = Name of symbol to test that resides in $(D T).
+ *
+ * Returns:
+ *     $(D true) iff $(D member) is static.
+ */
+template isStaticMember(T, string member)
+    if (__traits(hasMember, T, member))
+{
+    import std.meta : Alias;
+    alias sym = Alias!(__traits(getMember, T, member));
+
+    static if (__traits(getOverloads, T, member).length == 0)
+        enum bool isStaticMember = __traits(compiles, &sym);
+    else
+        enum bool isStaticMember = __traits(isStaticFunction, sym);
+}
+
+///
+unittest
+{
+    static struct S
+    {
+        static void sf() {}
+        void f() {}
+
+        static int si;
+        int i;
+    }
+
+    static assert( isStaticMember!(S, "sf"));
+    static assert(!isStaticMember!(S, "f"));
+
+    static assert( isStaticMember!(S, "si"));
+    static assert(!isStaticMember!(S, "i"));
+
+    static assert(!__traits(compiles, isStaticMember!(S, "hello")));
+}
+
+unittest
+{
+    static struct S
+    {
+        enum X = 10;
+        enum Y
+        {
+            i = 10
+        }
+        struct S {}
+        class C {}
+
+        static int sx = 0;
+        __gshared int gx = 0;
+
+        Y y;
+        static Y sy;
+
+        static void f() {}
+        static void f2() pure nothrow @nogc @safe {}
+
+        shared void g() {}
+
+        static void function() fp;
+        __gshared void function() gfp;
+        void function() fpm;
+
+        void delegate() dm;
+        static void delegate() sd;
+
+        void m() {}
+        final void m2() const pure nothrow @nogc @safe {}
+
+        inout(int) iom() inout { return 10; }
```

Comment:
some of these are not covered; if they are needed as declarations but never called please don't define